### PR TITLE
Fix release notes integrations section generation

### PR DESCRIPTION
**Pull request by serinamarie**
Ensures we can retrieve the latest release commits from integrations repos. 

- Add `get_latest_and_previous_tags` fn to correctly fetch and handle tags from GitHub
- Add `clean_integrations_changelog` fn to cleanup the integrations changelog

### CODE REVIEW

File: scripts/generate-release-notes.py [hunk 1]
```diff
@@ -105,6 +105,49 @@ def get_latest_repo_release_date(repo_org: str, repo_name: str) -> datetime:
...
+    """
+    Retrieves the latest and the previous release tags for the specified repository.
+    """
+    headers = {
+        "Authorization": f"Bearer {github_token}",
+        "Accept": "application/vnd.github.v3+json",
+    }
+    response = httpx.get(
+        f"https://api.github.com/repos/{repo_org}/{repo_name}/tags", headers=headers
+    )
```
Comment (bunchesofdonald) [inline scripts/generate-release-notes.py:120]: Probably not a big deal, but always nice to call `raise_for_status` so we get tracebacks if the call fails.

```suggestion
    )
    response.raise_for_status()
```

File: scripts/generate-release-notes.py [hunk 2]
```diff
@@ -105,6 +105,49 @@ def get_latest_repo_release_date(repo_org: str, repo_name: str) -> datetime:
...
+    tags = response.json()
+    if not tags:
+        raise Exception(f"No tags found for {repo_name}")
+
+    # Assuming tags are sorted in reverse chronological order, which GitHub usually does by default.
+    latest_tag = tags[0]["name"]
```
Comment (bunchesofdonald) [inline scripts/generate-release-notes.py:126]: Worth explicitly sorting them in case that changes on github's side?

**Review by bunchesofdonald — APPROVED** [review]
Thank you!

File: scripts/generate-release-notes.py (hunk 1 above)
Comment (serinamarie) [reply scripts/generate-release-notes.py:120]: sounds good!

**Pull request merged**